### PR TITLE
Website updates and add ZapdosTestApp

### PR DESCRIPTION
**Pull request by cticenhour**
This PR: 

- Updates the website infrastructure files due to changes in MooseDocs (syntax changes and the addition of SQA). 
- Adds `ZapdosTestApp` and associated infrastructure comparable to the current Stork. MooseDocs now automatically looks for test inputs under `test/tests`, so it seemed reasonable to update to current app practices and move our tests to the "expected" location. 

~~Making this a work-in-progress PR for now.~~

Refs #41

### CODE REVIEW

File: doc/content/development/contributing.md
Comment (cticenhour) [inline doc/content/development/contributing.md:5]: Remove extra "refer"

**Comment by cticenhour** [comment]
@csdechant, @keniley1, and @lindsayad I think this is ready for review. To better review the website changes, please pull my changes, build the site, and take a look locally: 
```
cd zapdos
make -j8
cd doc
./moosedocs.py build --serve
```
which will host the website on `http://127.0.0.1:8000`.  @keniley1, I'd also like you to note any funding acknowledgements I wasn't aware of (see the bottom of `doc/content/index.md`). 

Moving the `tests` directory aligns Zapdos with other current MOOSE codes, but will require a rebase and conflict resolution on any outstanding code changes after this is merged. Because of this, it would make sense to wait on merging this, but since it was ready I thought I would at least get it reviewed. 

One more thing to note - running `./moosedocs.py check --reports app`  on this branch runs a new report on Zapdos software quality! This command allows us to monitor how many stub documentation files and object class descriptions are still outstanding. As of `04/20/2021`, this is the output: 

```
zapdos WARNING
              log_default: 0
           log_stub_files: 189
      log_duplicate_files: 0
  log_missing_description: 33
     log_missing_markdown: 0
```

Lots of work left to do... 😅 Thanks for taking a look everyone!

**Comment by lindsayad** [comment]
Wow I love the coloring :smile: My only negative initial thought is that the text in the top menu bar is a little faint.
Screenshot from 2021-04-28 14-40-44
Also a little faint on the a-to-z page.
Screenshot from 2021-04-28 14-43-26
Otherwise I think this looks really awesome

**Comment by cticenhour** [comment]
@lindsayad What browser are you using? The font is the default from the MOOSE webpages (only changed the primary page color to red over blue). I'm on Firefox 88 on MacOS, and it's much bolder than this (though still a pretty thin font, admittedly). I'll look into changing the font, or making a modification to make it bolder.

**Comment by lindsayad** [comment]
I'm on PopOS with Firefox 85

**Comment by keniley1** [comment]
I tried this on Chrome version 90 on a Ubuntu 20.04 machine. The font is perfectly readable to me. In Firefox 88 it looks like what @lindsayad is seeing. 

The rest looks awesome, @cticenhour! Really cool. Eventually we should add an Examples tab with a few detailed tutorials, but I think that would be better once we have a more user-friendly input file.

**Comment by cticenhour** [comment]
@keniley1 Thanks! I agree - adding an "Examples" or "Tutorials" tab is needed. I'm hoping to take @csdechant's PDF tutorial and integrate it into the website, with needed updates for new syntax, etc. I'd like to identify what the specific "validation" cases are as well, since they'll probably make the best examples, and then give them their own documentation pages with implementation notes, references, tips, and brief results, maybe.

Before I forget, did you have any funding acknowledgements you needed to add to the homepage? 

Also, I will look into making a font update so that it's a bit clearer across browsers.

File: doc/content/development/contributing.md
Comment (cticenhour) [inline doc/content/development/contributing.md:116]:
```suggestion
## Reach out to a member of the Zapdos development team
```

**Comment by keniley1** [comment]
> Before I forget, did you have any funding acknowledgements you needed to add to the homepage?

Nope -- the only funding I had was through NSF. It's all there!

**Comment by cticenhour** [comment]
@lindsayad - finally got back to this and changed the weight of the font on the top bar and links to make them a little bolder. On my machine (macOS 11.2.1, Firefox 88 and Edge (Chromium) 91), they look like:

Screen Shot 2021-06-03 at 6 16 10 PM

Screen Shot 2021-06-03 at 6 16 32 PM

**Review by lindsayad — APPROVED** [review]
I just built the doc and looked at it in Firefox on Ubuntu, and such wow, such beautiful, crisp font I have never seen before

**Comment by cticenhour** [comment]
@lindsayad  thanks! 

@csdechant this will cause more rebase pain for you since you have new tests in your paper branch, so I will hold off on merging it for now. Though, I suppose I could add a temporary symlink in `test` and leave `tests` where it currently is to eliminate the conflicts.....might test how stable that is for the MooseDocs system.

**Comment by cticenhour** [comment]
@csdechant I've been seeing a serious uptick in traffic and clones of Zapdos since the early summer, so I want to go ahead and get this merged, and we can handle the mess with the rebasing later. I want to get updated website directions and README info up so that new users can see it, even if its incomplete. 

Updated the MOOSE submodule to the current master branch, otherwise it's the same as when Alex approved back in June. 

Could you review this?

**Review by csdechant — APPROVED** [review]

**Comment by cticenhour** [comment]
Thanks!

**Pull request merged**
